src/main/java/seedu/address/logic/commands/AddAllergyCommand.java

```diff
@@ -80,6 +77,34 @@ public CommandResult execute(Model model) throws CommandException {
         return new CommandResult(generateSuccessMessage(editedPerson));
     }
 
+    /**
+     * Retrieves the updated set of allergies for a person, ensuring no duplicate allergies are added.
+     *
+     * @param person The {@code Person} whose allergies are being updated.
+     * @return A {@code Set<Allergy>} containing the updated list of allergies for the person.
+     * @throws CommandException if there are any duplicate allergies in the provided set.
+     */
+    private Set<Allergy> getUpdatedAllergySet(Person person) throws CommandException {
+        Set<Allergy> updatedAllergySet = new HashSet<>(person.getAllergies());
+        List<Allergy> duplicateAllergies = new ArrayList<>();
+
+        for (Allergy allergy : allergies) {
+            if (!updatedAllergySet.add(allergy)) {
+                duplicateAllergies.add(allergy);
+            }
+        }
+
+        if (!duplicateAllergies.isEmpty()) {
+            String duplicates = duplicateAllergies.stream()
+                    .map(Allergy::getAllergy)
+                    .collect(Collectors.joining(", "));
+            throw new CommandException(String.format(MESSAGE_DUPLICATE_ALLERGY, duplicates));
+        }
```

Comment:
can merge the 2 streams into one and then test if the collected string is empty:

```java
        String duplicates = medCons.stream()
                                   .filter(medCon -> !updatedMedConSet.add(medCon))
                                   .map(MedCon::getMedCon)
                                   .collect(Collectors.joining(", "));
        if (!duplicates.isEmpty()) {
            throw new CommandException(String.format(MESSAGE_DUPLICATE_MEDCON, duplicates));
        }

```